aesara/link/numba/dispatch/elemwise.py

```diff
@@ -162,6 +162,53 @@ def create_vectorize_func(
     return elemwise_fn
 
 
+def create_guvectorize_func(
+    scalar_op_fn: Callable,
+    node: Apply,
+    identity: Optional[Any] = None,
+    **kwargs,
+) -> Callable:
+    r"""Create a guvectorized Numba function from a `Apply`\s Python function."""
+
+    signature_ = create_numba_signature(node, force_scalar=False)
+    signature = [(*signature_.args, *signature_.return_type.types)]
+
+    target = (
+        getattr(node.tag, "numba__vectorize_target", None)
+        or config.numba__vectorize_target
+    )
+
+    layout = f"{','.join(('()',) * len(node.inputs))}->{','.join(('()',) * len(node.outputs))}"
+    print(f"{signature=}, {layout=}")
+    numba_guvectorized_fn = numba.guvectorize(
+        signature,
+        layout,
+        identity=identity,
+        target=target,
+        fastmath=config.numba__fastmath,
+    )
+
+    input_names = [f"i{i}" for i in range(len(node.inputs))]
+    output_names = [f"o{i}" for i in range(len(node.outputs))]
+    gu_fn_name = "gu_func"
```

Comment:
Not familiar with the auto-naming strategy we have going on with Numba, are there any developer docs I can use as a reference?

Reply:
We should only need to make sure that the names we generate are fixed and don't clobber each other.  As long as you're generating all the names yourself everything should be fine; it's usually when you're using unknown names provided by something else that problems start to arise.

On a very related note, if we want certain type of caching to work (e.g. the kind that's based on hashes of source code), we'll need to clean up some old code that uses `Variable.name`, and anything else that could differ between equivalent graphs.  Since most of the unique name-based code was used to avoid `Variable.name` issues, we can probably drop all of it now.  In summary, it might be useful for debugging and readability, but it's not necessary and it can negatively affect caching, so don't worry about it.